### PR TITLE
Add Set.IteratorAt and Set.IteratorFrom

### PR DESCRIPTION
**Pull request by arv**
These two were missing for Set

### CODE REVIEW

**Comment by arv** [comment]
I need this for GraphQL...

File: go/types/set.go [hunk 1]
```diff
@@ -217,7 +217,21 @@ func (s Set) IterAll(cb setIterAllCallback) {
...
+func (s Set) IteratorAt(idx uint64) SetIterator {
+	return &setIterator{
+		cursor: newCursorAtIndex(s.seq, idx, false),
+		s:      s,
+	}
+}
+
+func (s Set) IteratorFrom(val Value) SetIterator {
+	return &setIterator{
+		cursor: newCursorAtValue(s.seq, val, false, false, false),
```
Comment (kalman) [inline go/types/set.go]: We should enable readahead for the iterators. I missed it in the map iterator review too, could you fix those?

File: go/types/set_iterator.go [hunk 2]
```diff
@@ -32,48 +32,42 @@ type SetIterator interface {
...
+	cursor       *sequenceCursor
+	currentValue Value
 }
 
 func (si *setIterator) Next() Value {
 	if si.cursor == nil {
-		si.cursor = newCursorAt(si.s.seq, emptyKey, false, false, false)
-	} else {
-		si.cursor.advance()
+		d.Panic("Cannot use a nil SetIterator")
```
Comment (kalman) [inline go/types/set_iterator.go]: Could you remove this condition and the panic, I don't think it's useful. Also in map iterator. I made that comment but Aaron didn't see it I suppose.

File: go/types/set_iterator.go [hunk 3]
```diff
@@ -32,48 +32,42 @@ type SetIterator interface {
...
+		si.cursor.advance()
+	} else {
+		si.currentValue = nil
 	}
-	return nil
+	return si.currentValue
 }
 
 func (si *setIterator) SkipTo(v Value) Value {
 	d.Chk.NotNil(v, "setIterator.SkipTo() called with nil value")
```
Comment (kalman) [inline go/types/set_iterator.go]: `d.PanicIfTrue(v == nil)`

**Comment by arv** [comment]
Thanks, will fix'em...

**Pull request merged**